src/Catalyst.Core.Modules.Ledger/Ledger.cs

```diff
@@ -48,31 +53,44 @@ namespace Catalyst.Core.Modules.Ledger
     public class Ledger : ILedger, IDisposable
     {
         public IAccountRepository Accounts { get; }
+        private readonly IDeltaExecutor _deltaExecutor;
+        private readonly IStateProvider _stateProvider;
+        private readonly IStorageProvider _storageProvider;
+        private readonly ISnapshotableDb _stateDb;
+        private readonly ISnapshotableDb _codeDb;
         private readonly ILedgerSynchroniser _synchroniser;
         private readonly IMempool<TransactionBroadcastDao> _mempool;
         private readonly ILogger _logger;
         private readonly IDisposable _deltaUpdatesSubscription;
 
         private readonly object _synchronisationLock = new object();
-        private readonly ICryptoContext _cryptoContext;
 
         public Cid LatestKnownDelta { get; private set; }
         public bool IsSynchonising => Monitor.IsEntered(_synchronisationLock);
 
-        public Ledger(IAccountRepository accounts,
+        public Ledger(IDeltaExecutor deltaExecutor,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Constructor has 10 parameters, which is greater than the 7 authorized.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4350015)